app/src/main/java/com/andrew/tasky/presentation/views/PasswordTextField.kt

```diff
@@ -2,20 +2,34 @@ package com.andrew.tasky.presentation.views
 
 import android.content.Context
 import android.util.AttributeSet
+import android.widget.Toast
 import com.andrew.tasky.R
 import com.andrew.tasky.databinding.TextFieldPasswordBinding
 import com.google.android.material.textfield.TextInputLayout
+import java.util.regex.Pattern
 
 
 class PasswordTextField @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null
 ): TextInputLayout(context, attrs) {
 
+    private val passwordRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{9,}\$"
+
+    private val passwordPattern = Pattern.compile(
+        passwordRegex
+    )
+
     private val binding = TextFieldPasswordBinding.bind(
         inflate(context, R.layout.text_field_password, this)
     )
 
-    init {
+    fun isPasswordValid(): Boolean{
```

Comment:
Personal opinion: Regex helps to implement validation patterns with little code, but it makes your code super hard to read. Nobody will be able to tell you from the pattern what the requirements for a valid password are. Something like this is a lot more readable:
```
class PasswordValidator {

    fun validate(password: String): ValidationResult {
        val containsLowerCase = password.any { it.isLowerCase() }
        val containsUpperCase = password.any { it.isUpperCase() }
        val containsNumber = password.any { it.isDigit() }
        if(password.length < MIN_PASSWORD_LENGTH || !containsLowerCase || !containsUpperCase || !containsNumber) {
            return ValidationResult(
                isSuccessful = false,
                error = UiText.StringResource(R.string.error_invalid_password_format, MIN_PASSWORD_LENGTH)
            )
        }

        return ValidationResult(isSuccessful = true)
    }

    companion object {
        const val MIN_PASSWORD_LENGTH = 9
    }
}
```

Regex is great, but for such easy checks I'd avoid it. It's great to check more complex patterns like an email address.